api/src/opentrons/protocol_api/module_contexts.py

```diff
@@ -770,3 +784,194 @@ def total_step_count(self) -> Optional[int]:
+        """
+        return self._module.speed_status.value
+
+    # TODO: add API version requirement
+    @property
+    def labware_latch_status(self) -> str:
+        """Heater-shaker's labware latch status string.
+
+        Returns one of these possible status values:
+        - "opening": latch is opening
+        - "idle_open": latch is open and idle
+        - "closing": latch is closing
+        - "idle_closed": latch is closed and idle
+        - "idle_unknown": status upon reset
+        - "unknown": latch status cannot be reached, likely due to an error
+        """
+        return self._module.labware_latch_status.value
+
+    # TODO: add API version requirement
+    def set_and_wait_for_temperature(self, celsius: float) -> None:
+        """Set and wait for target temperature.
+
+        Sets the heater-shaker target temperature and delays protocol execution
+        until target temperature has reached.
+
+        Note: The H/S truncates the temperature param to 2 decimal places
+
+        :param celsius: The target temperature, in °C in range 37°C to 95°C.
+        """
+        self.set_target_temperature(celsius=celsius)
+        self.wait_for_temperature()
+
+    # TODO: add API version requirement
+    @publish(command=cmds.heater_shaker_set_target_temperature)
+    def set_target_temperature(self, celsius: float) -> None:
+        """Set target temperature and return immediately.
+
+        Sets the heater-shaker's target temperature and returns immediately without
+        waiting for the target to be reached. Does not delay the protocol until
+        target temperature has reached. Use `wait_for_target_temperature` to delay
+        protocol execution.
+
+        Note: The H/S truncates the temperature param to 2 decimal places
+        """
+        validated_temp = validate_heater_shaker_temperature(celsius=celsius)
+        self._module.start_set_temperature(celsius=validated_temp)
+
+    # TODO: add API version requirement
+    @publish(command=cmds.heater_shaker_wait_for_temperature)
+    def wait_for_temperature(self) -> None:
+        """Wait for target temperature.
+
+        Delays protocol execution until heater-shaker has reached its target
+        temperature. The module should have a target temperature set previously.
+        """
+        if self.target_temperature is None:
+            raise NoTargetTemperatureSetError(
+                f"Heater-shaker module {self} does not have a target temperature set."
+            )
+        self._module.await_temperature(awaiting_temperature=self.target_temperature)
+
+    # TODO: add API version requirement
+    @publish(command=cmds.heater_shaker_set_and_wait_for_shake_speed)
+    def set_and_wait_for_shake_speed(self, rpm: int) -> None:
+        """Set and wait for target speed.
+
+        Set the heater shaker's target speed and wait until the specified speed has
+        reached. Delays protocol execution until the target speed has been achieved.
+        """
```

Comment:
```suggestion
        """Set a target shake speed and wait for it to be reached."""
```

For brevity. Feel free to disregard this suggestion if you think the more verbose version adds clarity.